libs/ui-lib/lib/ocm/components/clusterConfiguration/staticIp/components/FormViewHosts/FormViewHostsFields.tsx

```diff
@@ -1,38 +1,119 @@
 import React from 'react';
 import { FormGroup, Grid, TextContent, Text, TextVariants } from '@patternfly/react-core';
-import { useField } from 'formik';
+import { useField, useFormikContext } from 'formik';
 import StaticIpHostsArray, { HostComponentProps } from '../StaticIpHostsArray';
-import { getFieldId } from '../../../../../../common';
+import { getFieldId, PopoverIcon } from '../../../../../../common';
 import HostSummary from '../CollapsedHost';
 import { FormViewHost, StaticProtocolType } from '../../data/dataTypes';
 import { getProtocolVersionLabel, getShownProtocolVersions } from '../../data/protocolVersion';
 import { getEmptyFormViewHost } from '../../data/emptyData';
-import { OcmInputField } from '../../../../ui/OcmFormFields';
+import { OcmCheckboxField, OcmInputField } from '../../../../ui/OcmFormFields';
+import '../staticIp.css';
+import BondsSelect from './BondsSelect';
+import BondsConfirmationModal from './BondsConfirmationModal';
 
 const getExpandedHostComponent = (protocolType: StaticProtocolType) => {
   const Component: React.FC<HostComponentProps> = ({ fieldName, hostIdx }) => {
+    const [isModalOpen, setIsModalOpen] = React.useState<boolean>(false);
+    const { setFieldValue } = useFormikContext();
+    const [bondPrimaryField] = useField(`${fieldName}.bondPrimaryInterface`);
+    const [bondSecondaryField] = useField(`${fieldName}.bondSecondaryInterface`);
+
+    const [useBond, setUseBond] = React.useState<boolean>(
+      !!bondPrimaryField.value || !!bondSecondaryField.value,
+    );
```

Comment:
I suggest removing this variable entirely and using the `useBond` field value instead to show/hide the bond fields.